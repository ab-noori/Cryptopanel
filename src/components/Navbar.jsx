import React from 'react';
import { Link, useLocation } from 'react-router-dom';
import { useSelector } from 'react-redux';
import './Navbar.scss';

const Navbar = () => {
  const location = useLocation();
  const showFlashbackButton = location.pathname !== '/Cryptopanel/';
  const currencyName = useSelector((state) => state.CurrencyDetails.CurrencyDetails?.name);
  const pageTitle = location.pathname === '/Cryptopanel/' ? 'Currencies' : currencyName;

  return (
    <nav className="navbar">
      {showFlashbackButton && (
        <div className="navbar-left">
          <Link to="/Cryptopanel/" className="navbar-flashback">
            Back
          </Link>
        </div>
      )}
      <div className="navbar-title-container">
        <div className="navbar-title"><h2>{pageTitle}</h2></div>
      </div>
    </nav>
  );
};

export default Navbar;
