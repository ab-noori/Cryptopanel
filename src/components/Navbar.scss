.navbar {
    padding: 1rem;
    height: 4vh;
    display: flex;
    align-items: center;
    background-color: #e4e1e1;
    position: relative; /* Add relative positioning to the navbar */
  }
  
  .navbar-left {
    margin-right: auto;
    margin-left: 5%;
  }
  
  .navbar-title-container {
    position: absolute; /* Position the title absolutely */
    left: 50%; /* Position it in the center horizontally */
    transform: translateX(-50%); /* Adjust for center alignment */
  }
  
  .navbar-title {
    color: #333;
    font-size: 1.2rem;
  }

  .navbar-title h2 {
    margin: 0;
    white-space: nowrap;
  }
  
  .navbar-flashback {
    color: #fff;
    padding: 0.2rem 0.9rem 0.3rem;
    border-radius: 4px;
    text-decoration: none;
    background-color: #333;
    cursor: pointer;
  }
  
  @media screen and (max-width: 768px) {
    .navbar {
      justify-content: center;
    }

    .navbar-title {
      color: #333;
      font-size: 1.1rem;
      margin: auto 10px;
    }

    .navbar-left {
      margin-right: auto;
      margin-left: 0;
    }
  }
  