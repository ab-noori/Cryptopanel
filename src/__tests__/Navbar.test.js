import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { useSelector } from 'react-redux';
import Navbar from '../components/Navbar';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

describe('Navbar component', () => {
  beforeEach(() => {
    useSelector.mockClear();
  });

  test('renders correctly when on Cryptopanel page', () => {
    useSelector.mockReturnValue({
      CurrencyDetails: {
        name: 'Bitcoin',
        symbol: 'BTC',
      },
    });

    const { container } = render(
      <Router>
        <Navbar />
      </Router>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly when not on Cryptopanel page', () => {
    useSelector.mockReturnValue({
      CurrencyDetails: {
        name: 'Ethereum',
        symbol: 'ETH',
      },
    });

    const { container } = render(
      <Router>
        <Navbar />
      </Router>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
