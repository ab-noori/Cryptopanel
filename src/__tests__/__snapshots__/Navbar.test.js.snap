// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders correctly when not on Cryptopanel page 1`] = `
<nav
  class="navbar"
>
  <div
    class="navbar-left"
  >
    <a
      class="navbar-flashback"
      href="/Cryptopanel/"
    >
      Back
    </a>
  </div>
  <div
    class="navbar-title-container"
  >
    <div
      class="navbar-title"
    >
      <h2>
        undefined (undefined)
      </h2>
    </div>
  </div>
</nav>
`;

exports[`Navbar component renders correctly when on Cryptopanel page 1`] = `
<nav
  class="navbar"
>
  <div
    class="navbar-left"
  >
    <a
      class="navbar-flashback"
      href="/Cryptopanel/"
    >
      Back
    </a>
  </div>
  <div
    class="navbar-title-container"
  >
    <div
      class="navbar-title"
    >
      <h2>
        undefined (undefined)
      </h2>
    </div>
  </div>
</nav>
`;
